api/app/serializers/v1/text_serializer.rb

```diff
@@ -134,7 +134,16 @@ def ingestion_external_source_url(object, params)
 
       def sections_map(object)
         sections_ids = object.spine & object.text_sections.pluck(:id)
-        sections_ids.map { |id| Hash[id: id.to_s, name: object.text_sections.find(id).name] }
+
+        sections_ids.map do |id|
+          section = object.text_sections.find(id)
+
+          {
+            id: id.to_s,
+            name: section.name,
+            hidden: section.hidden_in_reader
+          }
```

Comment:
This whole method is better expressed as:

```ruby
object.text_sections.where(id: object.spine).map do |text_section|
  text_section.slice(:id, :name).merge(hidden: text_section.hidden_in_reader)
 end
```

Reply:
Technically I might add a method like:

```ruby
def to_section_map
  slice(:id, :name).merge(hidden: hidden_in_reader)
end
```

to `TextSection`, then you can just have:

```ruby
object.text_sections.where(id: object.spine).map(&:to_section_map)
```